tracer/src/Datadog.Trace/AppSec/EventTrackingSdk.cs

```diff
@@ -17,172 +16,194 @@ namespace Datadog.Trace.AppSec;
         }
-
-        /// <summary>
-        /// Sets the details of a custom event the local root span
-        /// </summary>
-        /// <param name="eventName">the name of the event to be tracked</param>
-        /// <param name="metadata">Metadata associated with the custom event</param>
-        [PublicApi]
-        public static void TrackCustomEvent(string eventName, IDictionary<string, string> metadata)
+    }
+
+    /// <summary>
+    /// Sets the details of a custom event the local root span
+    /// </summary>
+    /// <param name="eventName">the name of the event to be tracked</param>
+    [PublicApi]
+    public static void TrackCustomEvent(string eventName)
+    {
+        TelemetryFactory.Metrics.Record(PublicApiUsage.EventTrackingSdk_TrackCustomEvent);
+        TrackCustomEvent(eventName, null, Tracer.Instance);
+    }
+
+    /// <summary>
+    /// Sets the details of a custom event the local root span
+    /// </summary>
+    /// <param name="eventName">the name of the event to be tracked</param>
+    /// <param name="metadata">Metadata associated with the custom event</param>
+    [PublicApi]
+    public static void TrackCustomEvent(string eventName, IDictionary<string, string> metadata)
+    {
+        TelemetryFactory.Metrics.Record(PublicApiUsage.EventTrackingSdk_TrackCustomEvent_Metadata);
+        TrackCustomEvent(eventName, metadata, Tracer.Instance);
+    }
+
+    internal static void TrackCustomEvent(string eventName, IDictionary<string, string> metadata, Tracer tracer)
+    {
+        if (string.IsNullOrEmpty(eventName))
         {
-            TelemetryFactory.Metrics.Record(PublicApiUsage.EventTrackingSdk_TrackCustomEvent_Metadata);
-            TrackCustomEvent(eventName, metadata, Tracer.Instance);
+            ThrowHelper.ThrowArgumentNullException(nameof(eventName));
         }
 
-        internal static void TrackCustomEvent(string eventName, IDictionary<string, string> metadata, Tracer tracer)
-        {
-            if (string.IsNullOrEmpty(eventName))
-            {
-                ThrowHelper.ThrowArgumentNullException(nameof(eventName));
-            }
+        var span = tracer.ActiveScope?.Span;
 
-            var span = tracer?.ActiveScope?.Span;
+        if (span is null)
+        {
+            ThrowHelper.ThrowException("Can't create a tracking event with no active span");
+        }
 
-            if (span is null)
-            {
-                ThrowHelper.ThrowException("Can't create a tracking event with no active span");
-            }
+        var setTag = TaggingUtils.GetSpanSetter(span, out var internalSpan);
 
-            var setTag = TaggingUtils.GetSpanSetter(span);
+        setTag(Tags.AppSec.Track(eventName), "true");
 
-            setTag(Tags.AppSec.Track(eventName), "true");
+        var eventPrefix = $"{Tags.AppSec.Events}{eventName}";
 
-            setTag($"_dd.{Tags.AppSec.Events}{eventName}.sdk", "true");
+        setTag($"_dd.{eventPrefix}.sdk", "true");
```

Comment:
OK, this time you definitely force extra allocations, I think it would be better to just have this (tiny) duplication in the string creation code

Reply:
thanks, fixed in [8fa918f](https://github.com/DataDog/dd-trace-dotnet/pull/6225/commits/8fa918f63e234e07613f17ed1634f12fdb342a0e)